schemas/validate-all-tools.py

```diff
@@ -73,6 +74,15 @@ def validate_tools():
             except json.JSONDecodeError as xc:
                 print("! JSON decoding error: file {}\n\t{}".format(filepath, xc))
                 sys.exit(2)
+            tool_name = json_data.get("name")
+            if json_data.get("id") is None:
+                failures.append(f"Null ID in tool: {tool_name}")
+            elif json_data.get("id") <= 0:
```

Comment:
This test presupposes that the ID is always numeric. Shouldn't it be preceded by a test checking if the ID is numeric?